p2p/swarm.go

```diff
@@ -307,16 +332,25 @@ func (s *swarm) sendMessageImpl(peerPubKey p2pcrypto.PublicKey, protocol string,
 
 	err = conn.Send(final)
 
-	s.lNode.Debug("Message sent successfully")
+	s.lNode.Debug("DirectMessage sent successfully")
 
 	return err
 }
 
-// RegisterProtocol registers an handler for `protocol`
-func (s *swarm) RegisterProtocol(protocol string) chan service.Message {
-	mchan := make(chan service.Message, config.ConfigValues.BufferSize)
+// RegisterDirectProtocol registers an handler for direct messaging based `protocol`
+func (s *swarm) RegisterDirectProtocol(protocol string) chan service.DirectMessage {
+	mchan := make(chan service.DirectMessage, config.ConfigValues.BufferSize)
+	s.protocolHandlerMutex.Lock()
+	s.directProtocolHandlers[protocol] = mchan
+	s.protocolHandlerMutex.Unlock()
+	return mchan
+}
+
+// RegisterGossipProtocol registers an handler for gossip based `protocol`
+func (s *swarm) RegisterGossipProtocol(protocol string) chan service.GossipMessage {
+	mchan := make(chan service.GossipMessage, config.ConfigValues.BufferSize)
```

Comment:
I know it was there already but it's better to take BufferSize from `s.config.BufferSize`. (where its loaded from config/flags)

Reply:
fixed